main.go

```diff
@@ -8,58 +8,32 @@ import (
 	"log"
 	"net/http"
 	"os"
+	"strconv"
 
 	"github.com/film42/pgreba/config"
 	"github.com/gorilla/handlers"
 	"github.com/gorilla/mux"
 )
 
+type MaxAllowableByteLagExceeded func(*http.Request, *NodeInfo) bool
+
 type HealthCheckWebService struct {
-	healthChecker *HealthChecker
+	healthChecker               *HealthChecker
+	maxAllowableByteLagExceeded MaxAllowableByteLagExceeded
 }
 
-// func (hc *HealthCheckWebService) getSlotHealthCheck(w http.ResponseWriter, r *http.Request) {
-// 	// Get request info
-// 	w.Header().Set("Content-Type", "application/json")
-// 	params := mux.Vars(r)
-// 	slotName := params["slot_name"]
-
-// 	// Perform the health check.
-// 	err := hc.healthChecker.CheckReplicationSlot(slotName)
-
-// 	// If the slot is OK, return status: ok.
-// 	if err == nil {
-// 		json.NewEncoder(w).Encode(map[string]string{
-// 			"status": "ok",
-// 			"slot":   slotName,
-// 		})
-// 		return
-// 	}
-
-// 	// If there was an error, set the appropriate status code.
-// 	switch err {
-// 	case ErrReplicationSlotNotFound:
-// 		w.WriteHeader(http.StatusNotFound)
-// 	case ErrReplicationSlotLagTooHigh:
-// 		w.WriteHeader(http.StatusServiceUnavailable)
-// 	default:
-// 		w.WriteHeader(http.StatusInternalServerError)
-// 	}
-
-// 	// Return error to the client.
-// 	json.NewEncoder(w).Encode(map[string]string{
-// 		"error": err.Error(),
-// 		"slot":  slotName,
-// 	})
-// }
-
 func (hc *HealthCheckWebService) apiGetIsPrimary(w http.ResponseWriter, r *http.Request) {
 	nodeInfo, err := hc.healthChecker.dataSource.GetNodeInfo()
 	if err != nil {
 		// Return a 500. Something bad happened.
 		panic(err)
 	}
 
+	// if byte lag exceeds max_allowable_byte_lag then return 500
```

Comment:
We don't need to check byte lag if it's a primary database.

Reply:
duh 🤦